decorator.rb

```diff
@@ -0,0 +1,13 @@
+require_relative 'nameable'
+
+# A decorator class with a correct_name method
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
```

Comment:
- [ ]  I will encourage you to access the nameable file through something like `attr_accessor :nameable`